File: Source/Run/SPA/src/app/models/task-item.ts
export interface TaskItem {
  id: string;
  investmentId: string;
  investmentName?: string;
  institutionName?: string;
  title: string;
  description?: string;
  dueYear: number;
  dueMonth: number;
  dueDay?: number;
  isCompleted: boolean;
  completedDate?: string;
  requiredAmount?: number;
  /** BRL, USD, EUR — from investment */
  currency?: string;
  createdDate: string;
}

export interface FutureTask {
  title: string;
  description?: string;
  investmentName?: string;
  institutionName?: string;
  dueYear: number;
  dueMonth: number;
  dueDay?: number;
  amount?: number;
  currency?: string;
  taskType: string; // 'Maturity' | 'MonthlyMovement'
}

File: Source/_Core/Contracts/Domain/TaskItem.cs
namespace WealthLedger.Contracts.Domain;

public class TaskItem : BaseEntity
{
    public Guid InvestmentId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int DueYear { get; set; }
    public int DueMonth { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime? CompletedDate { get; set; }
}
